registry/storage/filewriter.go

```diff
@@ -22,6 +23,11 @@ type fileWriter struct {
 	err    error // terminal error, if set, reader is closed
 }
 
+type bufferedFileWriter struct {
+	*fileWriter
```

Comment:
As is, if you just change the embed declaration, offset on the embedded struct never gets updated even though I've confirmed fw.offset has been updated inside readFromAt.